test/unit/salt_test_cli_test.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+require 'webmock/test_unit'
+require 'mocha/test_unit'
+require 'rack/test'
+require 'tempfile'
+
+require 'smart_proxy_salt/salt'
+require 'smart_proxy_salt/salt_api'
+
+class SaltCLITest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    Proxy::Salt::Api.new
+  end
+
+  def setup
+    @salt_rest_api = 'http://localhost/salt_rest_api'
+
+    autosign_file = Tempfile.new('autosign_file')
```

Comment:
Not completely sure if this can be a problem here:
The documentation says that files created by `Tempfile` will be automatically deleted if the application exits **or the object is garbage-collected** (see [Class: Tempfile](https://ruby-doc.org/stdlib-2.5.3/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close))

So you might better define `autosign_file` as object-attribute instead using `@autosign_file` to makes sure a reference to the object exists at least as long as the instance of the test-class exists.

Reply:
Shouldn't be a problem because setup is called before each test - and, honestly, I'm pretty happy that the Tempfiles are garabe-collected afterwards.